clarity/src/vm/analysis/type_checker/mod.rs

```diff
@@ -310,19 +314,54 @@ impl FunctionType {
         for (expected_arg, arg) in expected_args.iter().zip(func_args.iter()).into_iter() {
             match (&expected_arg.signature, arg) {
                 (
-                    TypeSignature::TraitReferenceType(trait_id),
+                    TypeSignature::TraitReferenceType(expected_trait_id),
                     Value::Principal(PrincipalData::Contract(contract)),
                 ) => {
                     let contract_to_check = db
                         .load_contract(contract)
                         .ok_or_else(|| CheckErrors::NoSuchContract(contract.name.to_string()))?;
-                    let trait_definition = db
-                        .get_defined_trait(&trait_id.contract_identifier, &trait_id.name)
+                    let expected_trait = db
+                        .get_defined_trait(
+                            &expected_trait_id.contract_identifier,
+                            &expected_trait_id.name,
+                        )
+                        .unwrap()
```

Comment:
This was there before, so I presumed it was safe, but you're right, it looks like it should instead use a `?` because `get_defined_trait` can return a `NoSuchContract` error. Thanks for catching that. I should've checked.

Reply:
Addressed in e59e8754b